templates/types/streaming/nextjs/app/components/ui/chat/chat-input.tsx

```diff
@@ -121,8 +122,11 @@ export default function ChatInput(
             disabled: props.isLoading,
           }}
         />
-        {isUsingLLamaCloud && (
-          <LlamaCloudSelector projects={projects} setPipeline={setPipeline} />
+        {process.env.NEXT_PUBLIC_USE_LLAMACLOUD === "true" && (
+          <LlamaCloudSelector
+            requestData={requestData}
```

Comment:
why do we need to pass requestData? Shouldn't that be just read only?

Reply:
got it, i think if we need to merge it, that should be done in chat-input, I'll change it